### ISSUE TITLE
Media sanity check failed. Counts are off.

### ISSUE THREAD

**Issue by hssm** [ankidroid/Anki-Android#2247]
Originally reported on Google Code with ID 1345

```
What steps will reproduce the problem?
Not sure, happens during media syncing.

It means that after media syncing we have the wrong number of media files between ankiweb
and AnkiDroid

Which means that at least one of the following cases is not handled properly by Ankidroid
when media syncing:
- Add a media file in anki, means that while syncing AnkiDroid didn't manage to create
the file locally.
- Delete a media file in anki, means that while syncing AnkiDroid didn't manage to
delete the file locally.
- Add a media file in AnkiDroid, means that after syncing the server the server didn't
create the new file correctly.
- Delete a media file from AnkiDroid, means that after syncing the server hasn't deleted
the file.

The problem could be due to local FS issues not handled properly, or not sending the
right instructions to ankiweb when syncing (eg asking to delete wrong filenames that
don't exist).

Or due to hidden files in media folder not being picked up during syncing, but still
counted in the sanity check.

In other words this needs lots of investigation.

It's currently the most frequent problem in beta16 and although it doesn't cause crashes,
it triggers full media syncs which can consume a lot of bandwith both for the users
and ankiweb.

So it needs resolving.
```

Reported by `inigo.aldana` on 2012-08-28 12:28:12
